packages/app/pages/organization/course/[cid]/edit.tsx

```diff
@@ -30,12 +30,15 @@ export default function Edit(): ReactElement {
   const { cid } = router.query
 
   const { organization } = useOrganization(profile?.organization.id)
+  const isAdmin =
+    profile && profile.organization.organizationRole === OrganizationRole.ADMIN
 
-  if (
-    profile &&
-    profile.organization.organizationRole !== OrganizationRole.ADMIN
-  ) {
-    return <DefaultErrorPage statusCode={404} />
+  const isProfessorInCourse = profile?.courses.some(
+    (course) => course.role === 'professor' && course.course.id === Number(cid),
+  )
+
+  if (!isAdmin && !isProfessorInCourse) {
```

Comment:
This should be probably in useEffect, there might be a race condition before page loads and API callback receives this information from backend?

Reply:
I don't understand